exporters/otlp/internal/otlpconfig/envconfig.go

```diff
@@ -145,6 +163,19 @@ func (e *EnvOptionsReader) GetOptionsFromEnv() []GenericOption {
 	return opts
 }
 
+func isInsecureEndpoint(v string) bool {
+	return strings.HasPrefix(strings.ToLower(strings.TrimSpace(v)), "http://")
```

Comment:
`isInsecureEndpoint` trims the whitespaces while `trimSchema` is not doing it
we should agree if we want to do it or not (I am neutral, but we should be consistent)
I would remove `strings.TrimSpace` from here OR put `v = strings.TrimSpace(v)` after `getEnvValue` call

Reply:
I agree and think it's best to delete it here. Not only because of the test but also that i think it's kind of a missconfiguration to add leading or trailing spaces to the endpoint value - it's also not supported in other environment variables like `COMPRESSION`

Reply:
sorry, that's wrong. It's already trimmed for all environment variables in the function `getEnvValue` (https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/internal/otlpconfig/envconfig.go#L150) 

But the outcome is the same - it's ok to delete it here.

Reply:
Someone resolve please 😉

Reply:
i introduced a test case for the with leading and trailing spaces.